src/dune_util/report_error.ml

```diff
@@ -135,6 +135,21 @@ let format_memo_stack pps =
                    (Pp.seq (Pp.verbatim "-> ")
                       (Pp.seq (Pp.text "required by ") pp))))))
 
+(* [print_separator ()] does nothing the first time it is called,
+   whereas subsequent calls print a new line. *)
+let print_separator =
+  let first_error = ref true
+  and msg =
+    { User_message.paragraphs = [ Pp.cut ]
+    ; hints = []
+    ; annots = User_message.Annots.empty
+    ; loc = None
+    }
+  in
+  fun () ->
+    if !first_error then first_error := false
```

Comment:
The reference never gets reset back to `true`, so I guess this feature doesn't interact well with file-watching builds. You'll need to add and use a mechanism for correctly handling rebuilds.